### PR TITLE
Undo reset page on sort apply, different action fetch data request for different pagination mode (#85).

### PR DESCRIPTION
**Pull request by AndrewRAM33**
Change sort behavior. When add sort, change page to first is canceled. 
Undo reset page on sort apply,  added different action fetch data request for different pagination mode.

### CODE REVIEW

File: src/actions/actions.ts [hunk 1]
```diff
@@ -121,13 +121,15 @@ export class ActionPayloadTypes {
...
+     * @param paginationMode From first to current page
      */
     bcFetchDataRequest: {
         bcName: string,
         depth?: number,
         widgetName: string,
         ignorePageLimit?: boolean,
-        keepDelta?: boolean
+        keepDelta?: boolean,
+        paginationMode?: boolean
```
Comment (Dergash) [inline src/actions/actions.ts]: 1) We should try to keep actions simple and directly connected to user interaction.  
Different use cases equals different actions, as your can see `bcFetchDataRequest` action is already overloaded with totally unintuitive parameters `depth`, `ignorePageLimit` and `keepDelta`, and they'll keep piling up.
I think we need to introduce a new action instead, something like:
```tsx
/**
 * Fetch data request request for specific pages range
 */
bcFetchDataPages: {
    /**
     * The business component to fetch data for
     * @deprecated TODO: 2.0.0 Should be removed in favor of widgetName
     */
    bcName: string,
    /**
     * Fisrt page to fetch (default is 1)
     */
    from?: number,
    /**
     * Last page to fetch (default is current page)
     */
    to?: number
} = z
```
then add this action to epic
```tsx
const bcFetchDataEpic: Epic = (action$, store) => action$.ofType(
    types.bcFetchDataRequest,
    types.bcFetchDataPages,
    // ...other actions
)
```
then 
```tsx
const fetchParams: Record<string, any> = {
    _page: page,
    _limit: limit
}
if (action.type === types.bcFetchDataPages) {
    fetchParams._page = action.payload.from || 1
    fetchParams._limit = (action.payload.to || page - fetchParams._page) * limit
}
```

2) `paginationMode` property name suggests that it'll be either enum or union of some string constants.  
It is difficult to grasp what exactly will happen if this option is active without looking into implementation. So the property shouldn't be boolean or should be named like "allCurrentPages`

File: src/components/ColumnTitle/ColumnSort.tsx [hunk 2]
```diff
@@ -60,12 +60,9 @@ function mapDispatchToProps(dispatch: Dispatch) {
...
             dispatch($do.bcAddSorter({bcName, sorter}))
-            dispatch($do.bcForceUpdate({bcName}))
-            let currentPage = 1
-            while (currentPage < page) {
-                currentPage++
-                dispatch($do.bcLoadMore({bcName}))
-            }
+            page > 1
+                ? dispatch($do.bcForceUpdate({bcName: bcName, paginationMode: true}))
+                : dispatch($do.bcForceUpdate({bcName: bcName}))
```
Comment (Dergash) [inline src/components/ColumnTitle/ColumnSort.tsx]: It's simplier to put condition as property value:
```tsx
dispatch($do.bcForceUpdate({ bcName, paginationMode: page > 1 })
```

File: src/actions/actions.ts (hunk 1 above)
Comment (AndrewRAM33) [reply src/actions/actions.ts]: resolved in another PR (#102)

File: src/components/ColumnTitle/ColumnSort.tsx (hunk 2 above)
Comment (AndrewRAM33) [reply src/components/ColumnTitle/ColumnSort.tsx]: Сhanged dispatcher. Another action added.

**Pull request merged**